composer/trainer/trainer.py

```diff
@@ -1069,7 +1072,7 @@ def __init__(
         deepspeed_config: Optional[dict[str, Any]] = None,
         fsdp_config: Optional[dict[str, Any]] = None,
         fsdp_auto_wrap: bool = True,
-        parallelism_config: Optional[dict[str, Any]] = None,
+        parallelism_config: Optional[Union[dict[str, Any], ParallelismConfig]] = None,
```

Comment:
Why support dict as well?

Reply:
It's nice to have the option to do both imo -- dont need to force using our classes

Reply:
yeah i mean you're unpacking the dict anyways and instantiating a ParallelismConfig, so this seems fine